x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/comparison_side/get_subfield_changes/index.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { DiffableAllFields } from '../../../../../../../../common/api/detection_engine';
+import { stringifyToSortedJson } from '../utils';
+import { getSubfieldChangesForDataSource } from './data_source';
+import { getSubfieldChangesForKqlQuery } from './kql_query';
+import { getSubfieldChangesForEqlQuery } from './eql_query';
+import { getSubfieldChangesForEsqlQuery } from './esql_query';
+import { getSubfieldChangesForThreatQuery } from './threat_query';
+import { getSubfieldChangesForRuleSchedule } from './rule_schedule';
+import { getSubfieldChangesForRuleNameOverride } from './rule_name_override';
+import { getSubfieldChangesForTimestampOverride } from './timestamp_override';
+import { getSubfieldChangesForTimelineTemplate } from './timeline_template';
+import { getSubfieldChangesForBuildingBlock } from './building_block';
+import { getSubfieldChangesForThreshold } from './threshold';
+import type { SubfieldChange } from '../types';
+
+/**
+ * Splits a field into subfields and returns the changes between the old and new subfield values.
+ *
+ * @param fieldName - The name of the field for which subfield changes are to be computed.
+ * @param oldFieldValue - The old value of the field.
+ * @param newFieldValue - The new value of the field.
+ * @returns - An array of subfield changes.
+ */
+export const getSubfieldChanges = <FieldName extends keyof DiffableAllFields>(
+  fieldName: FieldName,
+  oldFieldValue?: DiffableAllFields[FieldName],
+  newFieldValue?: DiffableAllFields[FieldName]
+): SubfieldChange[] => {
+  switch (fieldName) {
+    /* Typecasting here because narrowing a union type (`DiffableAllFields[FieldName]`) is not supported by TS */
```

Comment:
```suggestion
    /* Typecasting `oldFieldValue` and `newFieldValue` to corresponding field type `DiffableAllFields[*]` is required here since `oldFieldValue` and `newFieldValue` concrete types depend on `fieldName` but TS isn't track that. */
```